### PR TITLE
Add table truncate feature

### PR DESCRIPTION
**Pull request by trevorsibanda**
Added table truncate feature
```scala
schema.truncate
```
#1029

### CODE REVIEW

**Comment by trevorsibanda** [comment]
Oracle tests causing travis-ci tests to fail

File: slick/src/main/scala/slick/jdbc/JdbcStatementBuilderComponent.scala [hunk 1]
```diff
@@ -589,13 +589,14 @@ trait JdbcStatementBuilderComponent { self: JdbcProfile =>
       if(primaryKeys.size > 1)
         throw new SlickException("Table "+tableNode.tableName+" defines multiple primary keys ("
           + primaryKeys.map(_.name).mkString(", ") + ")")
-      DDL(createPhase1, createPhase2, dropPhase1, dropPhase2)
+      DDL(createPhase1, createPhase2, dropPhase1, dropPhase2 , truncatePhase)
```
Comment (cvogt) [inline slick/src/main/scala/slick/jdbc/JdbcStatementBuilderComponent.scala]: why did you add this to the DDL phases?

File: slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/ActionTest.scala [hunk 2]
```diff
@@ -172,4 +172,24 @@ class ActionTest extends AsyncTest[RelationalTestDB] {
...
+
+    val ts = TableQuery[T]
+    for{
+      _ <- ts.schema.create
+      initial <- ts.result
+      _ = assert(initial.toSet == Set())
+      res <- (ts ++= Seq(2, 3, 1, 5, 4)) >>
+             ts.result
+      _ = assert(res.toSet == Set(2, 3, 1, 5, 4))
+      newRes <- ts.schema.truncate >>
```
Comment (cvogt) [inline slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/ActionTest.scala]: why is truncate an operation on the schema, not the table? it only modifies contents, not structure, so my intuition would put it into the table.

**Comment by cvogt** [comment]
LGTM

**Comment by cvogt** [comment]
cc @szeiger
